File: backend/app/schemas/claim_request.py
"""Pydantic schemas for claim requests (solicitudes)."""

from datetime import date, datetime, time
from decimal import Decimal
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class ClaimRequestOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    request_number: str | None
    status: str
    policyholder_id: UUID
    policy_id: UUID
    vehicle_id: UUID
    accident_date: date | None
    accident_time: time | None
    accident_location: str | None
    accident_lat: Decimal | None
    accident_lng: Decimal | None
    accident_description: str | None
    reported_damages: str | None
    submitted_at: datetime | None
    reviewed_by_user_id: UUID | None
    intake_decision_reason: str | None
    formalized_claim_id: UUID | None
    tenant_id: UUID
    created_at: datetime
    updated_at: datetime | None


class ClaimRequestCreate(BaseModel):
    policyholder_id: UUID
    policy_id: UUID
    vehicle_id: UUID
    accident_date: date | None = None
    accident_time: time | None = None
    accident_location: str | None = None
    accident_lat: Decimal | None = None
    accident_lng: Decimal | None = None
    accident_description: str | None = None
    reported_damages: str | None = None


class ClaimRequestUpdate(BaseModel):
    policyholder_id: UUID | None = None
    policy_id: UUID | None = None
    vehicle_id: UUID | None = None
    accident_date: date | None = None
    accident_time: time | None = None
    accident_location: str | None = None
    accident_lat: Decimal | None = None
    accident_lng: Decimal | None = None
    accident_description: str | None = None
    reported_damages: str | None = None


class ClaimRequestListResponse(BaseModel):
    items: list[ClaimRequestOut]
    total: int
    page: int
    limit: int


class RejectIntakePayload(BaseModel):
    reason: str = Field(min_length=20, max_length=2000)


class ReassignPayload(BaseModel):
    target_user_id: UUID

File: frontend/types/claim-request.ts
export interface ClaimRequest {
  id: string
  request_number: string | null
  status: string
  policyholder_id: string
  policy_id: string
  vehicle_id: string
  accident_date: string | null
  accident_time: string | null
  accident_location: string | null
  accident_lat: number | null
  accident_lng: number | null
  accident_description: string | null
  reported_damages: string | null
  submitted_at: string | null
  reviewed_by_user_id: string | null
  intake_decision_reason: string | null
  formalized_claim_id: string | null
  tenant_id: string
  created_at: string
  updated_at: string | null
}

export interface ClaimRequestListResponse {
  items: ClaimRequest[]
  total: number
  page: number
  limit: number
}

export interface RejectIntakePayload {
  reason: string
}

export interface ReassignPayload {
  target_user_id: string
}
